src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -782,4 +792,20 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
         }
         return result;
     }
+    
+    public List<String> getArchivedCourseIds(List<CourseDetailsBundle> allCourses, List<InstructorAttributes> instructorList) {
+        List<String> activeCourseIds = new ArrayList<String>();
+        for (CourseDetailsBundle cdb : allCourses) {
+            if (cdb.course.isArchived) {
```

Comment:
Actually I think in this pull request, the checks are still required, until we can be sure that all the course's isArchived status has been migrated to the instructors. I can remove them from CoursesLogic in #3747 next milestone.